### PR TITLE
add progressing status unit test and updated messages

### PR DESCRIPTION
**Pull request by sanchezl**

### CODE REVIEW

File: pkg/operator/workload_controller_openshiftapiserver_v311_00.go [hunk 1]
```diff
@@ -111,8 +114,17 @@ func syncOpenShiftAPIServer_v311_00_to_latest(c OpenShiftAPIServerOperator, orig
 	}
 
 	// If the daemonset is up to date and the operatorConfig are up to date, then we are no longer progressing
-	if actualDaemonSet.ObjectMeta.Generation == actualDaemonSet.Status.ObservedGeneration &&
-		operatorConfig.ObjectMeta.Generation == operatorConfig.Status.ObservedGeneration {
+	var progressingMessages []string
+	if actualDaemonSet.ObjectMeta.Generation != actualDaemonSet.Status.ObservedGeneration {
+		glog.Warningf("daemonset/apiserver.openshift-operator: observed generation is %d, actual generation is %d.", actualDaemonSet.Status.ObservedGeneration, actualDaemonSet.ObjectMeta.Generation)
```
Comment (deads2k) [inline pkg/operator/workload_controller_openshiftapiserver_v311_00.go]: if it's important enough to log, its important enough for an event.

Comment (deads2k) [inline pkg/operator/workload_controller_openshiftapiserver_v311_00.go]: this isn't a warning.  V(2) at best.

File: pkg/operator/workload_controller_openshiftapiserver_v311_00.go [hunk 2]
```diff
@@ -111,8 +114,17 @@ func syncOpenShiftAPIServer_v311_00_to_latest(c OpenShiftAPIServerOperator, orig
...
+		progressingMessages = append(progressingMessages, "The operator is process of updating a DaemonSet")
+	}
+	if operatorConfig.ObjectMeta.Generation != operatorConfig.Status.ObservedGeneration {
+		glog.Warningf("openshiftapiserveroperatorconfigs/instance: observed generation is %d, actual generation is %d.", operatorConfig.Status.ObservedGeneration, operatorConfig.ObjectMeta.Generation)
```
Comment (deads2k) [inline pkg/operator/workload_controller_openshiftapiserver_v311_00.go]: V(2)

File: pkg/operator/workload_controller_openshiftapiserver_v311_00.go [hunk 3]
```diff
@@ -122,9 +134,10 @@ func syncOpenShiftAPIServer_v311_00_to_latest(c OpenShiftAPIServerOperator, orig
 			Type:    operatorv1.OperatorStatusTypeProgressing,
 			Status:  operatorv1.ConditionTrue,
 			Reason:  "DesiredStateNotYetAchieved",
-			Message: fmt.Sprintf("Generation: expected: %v, actual: %v", operatorConfig.Status.ObservedGeneration, actualDaemonSet.ObjectMeta.Generation),
+			Message: strings.Join(progressingMessages, "\n"),
```
Comment (deads2k) [inline pkg/operator/workload_controller_openshiftapiserver_v311_00.go]: this is fine.

File: pkg/operator/workload_controller_openshiftapiserver_v311_00_test.go [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+	defer func() {
+		if r := recover(); r != nil {
+			t.Errorf("Panic: %v", r)
+			t.Error(string(debug.Stack()))
+		}
+	}()
+	f()
+}
+
+type mockRecorder struct {
```
Comment (deads2k) [inline pkg/operator/workload_controller_openshiftapiserver_v311_00_test.go]: Move this to a subpackage under spot where the in memory one is.  `TestingEventRecorder` and `NewTestingEventRecorder`

Comment (sanchezl) [reply pkg/operator/workload_controller_openshiftapiserver_v311_00_test.go]: Opened issue and will use `InMemoryRecorder` for now.

File: pkg/operator/workload_controller_openshiftapiserver_v311_00.go [hunk 5]
```diff
@@ -111,8 +112,15 @@ func syncOpenShiftAPIServer_v311_00_to_latest(c OpenShiftAPIServerOperator, orig
 	}
 
 	// If the daemonset is up to date and the operatorConfig are up to date, then we are no longer progressing
-	if actualDaemonSet.ObjectMeta.Generation == actualDaemonSet.Status.ObservedGeneration &&
-		operatorConfig.ObjectMeta.Generation == operatorConfig.Status.ObservedGeneration {
+	var progressingMessages []string
+	if actualDaemonSet.ObjectMeta.Generation != actualDaemonSet.Status.ObservedGeneration {
+		progressingMessages = append(progressingMessages, fmt.Sprintf("daemonset/apiserver.openshift-operator: observed generation is %d, actual generation is %d.", actualDaemonSet.Status.ObservedGeneration, actualDaemonSet.ObjectMeta.Generation))
```
Comment (deads2k) [inline pkg/operator/workload_controller_openshiftapiserver_v311_00.go]: observed generation is, desired generation is

File: pkg/operator/workload_controller_openshiftapiserver_v311_00.go [hunk 6]
```diff
@@ -111,8 +112,15 @@ func syncOpenShiftAPIServer_v311_00_to_latest(c OpenShiftAPIServerOperator, orig
...
+	}
+	if operatorConfig.ObjectMeta.Generation != operatorConfig.Status.ObservedGeneration {
+		progressingMessages = append(progressingMessages, fmt.Sprintf("openshiftapiserveroperatorconfigs/instance: observed generation is %d, actual generation is %d.", operatorConfig.Status.ObservedGeneration, operatorConfig.ObjectMeta.Generation))
```
Comment (deads2k) [inline pkg/operator/workload_controller_openshiftapiserver_v311_00.go]: observed and desired, not actual.  They are both actual

File: pkg/operator/workload_controller_openshiftapiserver_v311_00_test.go [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+			if condition.Message != tc.expectedMessage {
+				t.Errorf("expected message:\n%v\nactual message:\n%v", tc.expectedMessage, condition.Message)
+			}
+
+		})
+	}
+
+}
+
+func assertNotPanic(t *testing.T, f func()) {
```
Comment (deads2k) [inline pkg/operator/workload_controller_openshiftapiserver_v311_00_test.go]: panics already fail tests

**Comment by deads2k** [comment]
nits, then lgtm

**Pull request merged**